include/bmx280.h

```diff
@@ -11,114 +11,136 @@
 #define _BMX280_H_
 
 #ifdef __cplusplus
-extern "C" {
+extern "C"
+{
 #endif
 
 #include <stdint.h>
 #include <limits.h>
-#include "driver/i2c.h"
+#include <freertos/FreeRTOS.h> //For vTaskDelay to delay after sensor reset
 #include "sdkconfig.h"
 
-#define BMXAPI extern
+#define USE_I2C_MASTER_DRIVER 1
 
 /**
- * Anonymous structure to driver settings.
+ * The ESP-IDF version 5.3.1 guide state that the legacy i2c.h driver and the i2c_master.h driver CANNOT coexist
  */
-typedef struct bmx280_t bmx280_t;
-
-#include "bmx280_bits.h"
+#ifdef USE_I2C_MASTER_DRIVER
+#include "driver/i2c_master.h"
+#else
+#include "driver/i2c.h"
+#endif
 
-/**
- * Create an instance of the BMX280 driver.
- * @param port The I2C port to use.
- * @return A non-null pointer to the driver structure on success.
- */
-BMXAPI bmx280_t* bmx280_create(i2c_port_t port);
-/**
- * Destroy your the instance.
- * @param bmx280 The instance to destroy.
- */
-BMXAPI void bmx280_close(bmx280_t* bmx280);
+#define BMXAPI extern
 
-/**
- * Probe for the sensor and read calibration data.
- * @param bmx280 Driver structure.
- */
-BMXAPI esp_err_t bmx280_init(bmx280_t* bmx280);
-/**
- * Configure the sensor with the given parameters.
- * @param bmx280 Driver structure.
- * @param configuration The configuration to use.
- */
-BMXAPI esp_err_t bmx280_configure(bmx280_t* bmx280, bmx280_config_t *cfg);
+    /**
+     * Anonymous structure to driver settings.
+     */
+    typedef struct bmx280_t bmx280_t;
 
-/**
- * Set the sensor mode of operation.
- * @param bmx280 Driver structure.
- * @param mode The mode to set the sensor to.
- */
-BMXAPI esp_err_t bmx280_setMode(bmx280_t* bmx280, bmx280_mode_t mode);
-/**
- * Get the sensor current mode of operation.
- * @param bmx280 Driver structure.
- * @param mode Pointer to write current mode to.
- */
-BMXAPI esp_err_t bmx280_getMode(bmx280_t* bmx280, bmx280_mode_t* mode);
+#include "bmx280_bits.h"
 
+    /**
+     * Create an instance of the BMX280 driver.
+     * @param port The I2C port to use.
+     * @return A non-null pointer to the driver structure on success.
+     */
+    BMXAPI bmx280_t *bmx280_create(i2c_port_t port);
+    /**
+     * Destroy your the instance.
+     * @param bmx280 The instance to destroy.
+     */
+    BMXAPI void bmx280_close(bmx280_t *bmx280);
+
+#if USE_I2C_MASTER_DRIVER
+    /**
+     * Probe for the senors and read calibration data.
+     * @param bmx280 Driver structure.
+     * @param dev_handle I2C device handle for BMX280 device, expected to be initialized by caller.
+     */
+    BMXAPI esp_err_t bmx280_init(bmx280_t *bmx280, i2c_master_dev_handle_t dev_handle);
```

Comment:
We cannot have two distinct init functions like this. This is incredibly weird considering we expect `bmx280_create()` to create the driver instance, which takes the legacy `i2c_port_t` enum value for legacy i2c driver.

I think you should modify the new API like this and keep the original function signature of `bmx280_init()`:

```c
#if USE_I2C_MASTER_DRIVER
// Your new entry point.
BMXAPI bmx280_t bmx280_create_master(i2c_master_dev_handle_t dev_handle);
#else
// The original entry point but renamed.
BMXAPI bmx280_t bmx280_create_legacy(i2c_port_t port);
// legacy define for existing code bases
#define bmx280_create(port) bmx280_create_legacy((port));
#endif
```
I'd like to hear what you think.

Reply:
Good point. I like your approach and will implement it accordingly.